packages/shared/src/lib/core/wallet/actions/send/createEvmChainToEvmChainTransaction.ts

```
@@ -0,0 +1,105 @@
+import { IChain } from '@core/network/interfaces'
+import { IAccountState } from '@core/account/interfaces'
+import { buildEvmTransactionData, getIscpTransferSmartContractData } from '@core/layer-2/actions'
+import { ISC_MAGIC_CONTRACT_ADDRESS } from '@core/layer-2/constants'
+import { AssetType } from '@core/layer-2/enums'
+import { EvmTransactionData, TransferredAsset } from '@core/layer-2/types'
+import { getErc20TransferSmartContractData } from '@core/layer-2/utils'
+import { TokenStandard } from '@core/token/enums'
+import { IToken } from '@core/token/interfaces'
+
+import { SendFlowType } from '../../stores'
+import { SendFlowParameters } from '../../types'
+
+export function createEvmChainToEvmChainTransaction(
+    sendFlowParameters: SendFlowParameters,
+    chain: IChain,
+    account: IAccountState
+): Promise<EvmTransactionData | undefined> {
+    if (
+        !sendFlowParameters ||
+        sendFlowParameters.type === SendFlowType.NftTransfer ||
+        !sendFlowParameters.recipient?.address
+    ) {
+        return Promise.resolve(undefined)
```

Isn't it better to throw errors then to resolve to undefined?